Feature/stepper

ydomenjoud opened the pull request.

ydomenjoud committed:
add stepper

ydomenjoud committed:
add stepper

ydomenjoud committed:
Merge branch 'master' into feature/stepper

# Conflicts:
#	projects/kalidea/kaligraphi/src/lib/molecules/kal-molecules.module.ts
#	projects/kalidea/kaligraphi/src/styles/kaligraphi.sass
#	src/app/app.component.html
#	src/app/atoms/atoms-routing.module.ts
#	src/app/atoms/atoms.module.ts

ydomenjoud committed:
add stepper feature

Bellico commented on projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/README.md:
````diff
@@ -0,0 +1,23 @@
+</kal-stepper>
+```
+
+* Doit afficher la liste d'étape donnée
````
French ?

Bellico commented on projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-step-header/kal-step-header.component.ts:
```diff
@@ -0,0 +1,63 @@
+  /** Label of the given step. */
+  @Input() label: KalStepLabelDirective | string;
+
+
```
Remove space

Bellico commented on projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-step-header/kal-step-header.component.ts:
```diff
@@ -0,0 +1,63 @@
+    return this._element.nativeElement;
+  }
+
+
```
Remove space

Bellico commented on projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-step-header/kal-step-header.component.ts:
```diff
@@ -0,0 +1,63 @@
+
+  /** Returns the host HTML element. */
+  getHostElement() {
+    return this._element.nativeElement;
```
We have not defined as convention prefix  '_' for private variables; 
Should we?
Currently it is not

ydomenjoud replied:
this is inherited from CDK ...

Bellico replied:
What  ?  (I speak for everything)

MichaelGrosio commented on projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-step-header/kal-step-header.component.ts:
```diff
@@ -0,0 +1,63 @@
+  @Input() optional: boolean;
+
+  constructor(
+    private _focusMonitor: FocusMonitor,
```
change private properties label

MichaelGrosio commented on projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-step-header/kal-step-header.component.ts:
```diff
@@ -0,0 +1,63 @@
+  }
+
+  ngOnDestroy() {
+    this._focusMonitor.stopMonitoring(this._element.nativeElement);
```
change private properties label

MichaelGrosio commented on projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-stepper.component.ts:
```diff
@@ -0,0 +1,60 @@
+import { takeUntil } from 'rxjs/operators';
+
+import { KalStepComponent } from './kal-step/kal-step.component';
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
```
imports

MichaelGrosio commented on projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-stepper.component.spec.ts:
```diff
@@ -0,0 +1,127 @@
+import { KalStepComponent } from './kal-step/kal-step.component';
+import { KalStepLabelDirective } from './directives/kal-step-label.directive';
+import { KalStepHeaderComponent } from './kal-step-header/kal-step-header.component';
+import { KalStepperModule } from 'projects/kalidea/kaligraphi/src/lib/molecules/kal-stepper/kal-stepper.module';
```
imports

ydomenjoud committed:
reorder files to improve readability
remove useless directives ( previous and next )
improve readme documentation
fix pr comments

ydomenjoud committed:
Merge branch 'master' into feature/stepper
